[FZ Editor] Replace ItemsControls with GridViews

niels9001 opened the pull request:
## Summary of the Pull Request

**THIS PR SHOULD BE MERGED IN BEFORE: #11047**


This PR replaces the ItemsControls (showing the monitors and layouts) with a ModernWPF GridView. This comes with the following advantages:

- Keyboard navigation is way better now and inline with Windows standards (Tab for moving between GridViews, arrow to move around between layouts)
- Solving a bunch of narrator issues.
- Mouse-over visual now inline with WinUI.
- Right-mouse context menu added (since we no longer have the MouseDown event to trigger the layout)

Final result:
![AccesibilityFZeditor](https://user-images.githubusercontent.com/9866362/116825327-14a59100-ab8f-11eb-92dd-b9107020f1ef.gif)


## Quality Checklist

- [X] **Linked issue:** #10842, #10841, #10784
- [ ] **Communication:** I've discussed this with core contributors in the issue. 
- [ ] **Tests:** Added/updated and all pass
- [ ] **Installer:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Docs:** Added/ updated
- [ ] **Binaries:** Any new files are added to WXS / YML
   - [ ] No new binaries
   - [ ] [YML for signing](https://github.com/microsoft/PowerToys/blob/master/.pipelines/pipeline.user.windows.yml#L68) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/master/installer/PowerToysSetup/Product.wxs) for new binaries

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.

htcfreek commented:
@niels9001 
I think the context menu is not a good idea. I don't see any improvement by adding it and it introduce a ux element without any indication that it exists.

niels9001 commented:
> @niels9001
> 
> > Right-mouse context menu added (since we no longer have the MouseDown event to trigger the layout)
> 
> What mouse down event. Don't understand what you mean. Can you please explain it more.

We now have a ItemClicked event. Before, we used the MouseDown event - making it difficult to determine if it was the Left or Right mouse button.

> I think the context menu is not a good idea. I don't see any improvement by adding it and it introduce a ux element without any indication that it exists.

Every major OS has right-click context menu's... And it's very common across applications on Windows. Alarms & Clock has the same logic as well (context menu and buttons in the dialog).

![image](https://user-images.githubusercontent.com/9866362/116827297-87673a00-ab98-11eb-95c0-829e050cc2c3.png)
 
![image](https://user-images.githubusercontent.com/9866362/116827304-93eb9280-ab98-11eb-8bf5-8d7207ea407c.png)

![image](https://user-images.githubusercontent.com/9866362/116827313-a5cd3580-ab98-11eb-853e-035323f33fe5.png)

htcfreek commented:
> > @niels9001
> > 
> > > Right-mouse context menu added (since we no longer have the MouseDown event to trigger the layout)
> > 
> > What mouse down event. Don't understand what you mean. Can you please explain it more.
> 
> We now have a ItemClicked event. Before, we used the MouseDown event - making it difficult to determine if it was the Left or Right mouse button.

This is the reason of an issue I reported in the past. We closed it as normal behavior. 

> > I think the context menu is not a good idea. I don't see any improvement by adding it and it introduce a ux element without any indication that it exists.
> 
> Every major OS begs to differ :)... And it's very common across applications on Windows: Alarms & Clock has the same UX supporting both:
> 
> Right-click menu:
> ![image](https://user-images.githubusercontent.com/9866362/116827402-11af9e00-ab99-11eb-8273-66bf1fcb10e2.png)
> 
> Dialog top button:
> ![image](https://user-images.githubusercontent.com/9866362/116827419-2a1fb880-ab99-11eb-9e43-3146bc0c9df1.png)
> 
>

niels9001 commented:
> Should we then add an "edit zones" item to shorten click path.

That's an interesting suggestion! We can handle that in a separate PR - might be additional menu items that would be good to add!

enricogior commented:
@niels9001 
nice work!
The context menu is great addition and I also like the idea suggested by @htcfreek to add the "Edit zones" option (maybe in another PR). Btw, how do you activate the context menu from the keyboard?

niels9001 commented:
@enricogior @htcfreek Actually, adding this functionality was easier than expected :). The quick layout editing is really nice for speeding up the workflow! If you have a better label for it that would be nice.

Shift + F10 is the default keyboard shortcut for Windows to open up a ContextMenu. This does not work in this PR apparently, will look into that.

![GridViewEditor](https://user-images.githubusercontent.com/9866362/116979007-c0043200-acc4-11eb-980c-31db542d4229.gif)

enricogior commented:
@niels9001 
what about using just `Edit zones` instead of `Edit zones layout`?

htcfreek commented:
@niels9001 
Didn't know this shortcut for context menu. You save my day. 😂

Does context menu key works instead?

htcfreek commented:
@niels9001 , @enricogior 
Though about a way to bting rename function to context menu. But because we ned a additional simple ui for just renaming, I think this should be a seperate pr/issue.

Two implementation ideas:
1. Simple dialogue with input, save button and abort button.
2. Change name label in gridview into input and show x and checkmark button behind. (Like in wt on renaming color sheme.)

niels9001 commented:
@enricogior Good suggestion - I have updated the label and moved the ContextMenu to the ItemContainerStyle: using Shift+F10 will now show the ContextMenu :). The entire layout selection page is now keyboard accessible!

![KeyboardContextMenu](https://user-images.githubusercontent.com/9866362/116997177-c0102c00-acdc-11eb-8c6e-560538b105a6.gif)


@htcfreek Yeah that makes sense.. but requires more work and additional dialogs to be introduced so kipping it for this PR.

enricogior commented:
@niels9001 
awesome!

enricogior commented:
@niels9001 
OK to delete the branch?

niels9001 commented:
Yup, deleted it.
